rholang/src/main/scala/coop/rchain/rholang/interpreter/Substitute.scala

```diff
@@ -37,8 +38,8 @@ object Substitute {
 
   def apply[M[_], A](implicit ev: Substitute[M, A]): Substitute[M, A] = ev
 
-  def maybeSubstitute[M[+ _]: InterpreterErrorsM](term: Var)(implicit depth: Int,
-                                                             env: Env[Par]): M[Either[Var, Par]] =
+  def maybeSubstitute[M[+ _]: Sync](term: Var)(implicit depth: Int,
```

Comment:
@henrytill I'm keeping context bounds here because even  though `depth` and `env` are marked as `implicit` they are being passed explicitly and passing `Sync` instances implicitly will be a  pain.